metricbeat/mb/module/connector.go

```diff
@@ -70,6 +76,27 @@ func NewConnector(
 	}, nil
 }
 
+// UseMetricSetProcessors appends processors defined in metricset configuration to the connector properties.
+func (c *Connector) UseMetricSetProcessors(r metricSetRegister, moduleName, metricSetName string) error {
+	metricSetProcessors, err := r.ProcessorsForMetricSet(moduleName, metricSetName)
+	if err != nil {
+		return errors.Wrapf(err, "reading metricset processors failed (module: %s, metricset: %s)",
+			moduleName, metricSetName)
+	}
+
+	if metricSetProcessors == nil || len(metricSetProcessors.List) == 0 {
+		return nil // no processors are defined
+	}
+
+	procs := processors.NewList(nil)
+	procs.AddProcessors(*metricSetProcessors)
+	for _, p := range c.processors.List {
+		procs.AddProcessor(p)
+	}
+	c.processors = procs
```

Comment:
It should work as expected now, yes :+1:

Reply:
Thanks!